src/main/java/com/commercetools/sync/producttypes/utils/AttributeDefinitionUpdateActionUtils.java

```diff
@@ -63,57 +65,117 @@
     }
 
     /**
-     * Compares all the {@link EnumValue} and {@link LocalizedEnumValue} values of an {@link AttributeDefinition} and
-     * an {@link AttributeDefinitionDraft} and returns a list of {@link UpdateAction}&lt;{@link ProductType}&gt; as a
-     * result. If both the {@link AttributeDefinition} and the {@link AttributeDefinitionDraft} have identical
-     * enum values, then no update action is needed and hence an empty {@link List} is returned.
+     * Checks if both the supplied {@code oldAttributeDefinition} and {@code newAttributeDefinitionDraft} have an
+     * {@link AttributeType} that is either an {@link EnumAttributeType} or a {@link LocalizedEnumAttributeType} or
+     * a {@link SetAttributeType} with a subtype that is either an {@link EnumAttributeType} or a
+     * {@link LocalizedEnumAttributeType}.
+     *
+     * <p>The method compares all the {@link EnumValue} and {@link LocalizedEnumValue} values of the
+     * {@link AttributeDefinition} and the {@link AttributeDefinitionDraft} attribute types and returns a list of
+     * {@link UpdateAction}&lt;{@link ProductType}&gt; as a result. If both the {@link AttributeDefinition} and the
+     * {@link AttributeDefinitionDraft} have identical enum values, then no update action is needed and hence an empty
+     * {@link List} is returned.</p>
+     *
+     * <p>Note: This method expects the supplied {@link AttributeDefinition} and the {@link AttributeDefinitionDraft}
```

Comment:
Is it possible to check it and throw exception if they're not the same? I think this note is not very visible to user of the library.

Reply:
Well this method is not exposed anyway to the use of the library. This comment is mainly for a maintainer of the library. This method is only called from one place where the types are guaranteed to be identical.